[~guozhang] [~bchen225242] I ran this test like 500 times locally and it never failed. And we only had a single failure on Jenkins. Closing this for now as "cannot reproduce". If it happens again, we can just reopen.

[~mjsax] I saw this fail locally, although it was for _exactly_once_ not _exactly_once_beta ._ Not sure if it really exposes an actual bug or is just flaky? The line that timed out was given just 15 seconds

Saw this fail again locally, will submit a PR

Resolving for now. If the fix is not good enough, we can just reopen again.

